### DISCUSSION TITLE
How to stop  component reloading , when there are no change in api response.

### DISCUSSION THREAD

**Discussion by sanpanthera** [TanStack/query#851] [General]
Currently on window focus component is reloading after every api call. I want to reload the component on window focus only **when there is the change in data**.

**Comment by the-homeless-god** [comment]
@tannerlinsley could you please provide any solution here?

**Reply by the-homeless-god** [reply]
@sanpanthera you can find all requested documentation here https://react-query.tanstack.com/guides/window-focus-refetching

**Reply by the-homeless-god** [reply]
@sanpanthera 

short-answer:

- for global usage
```javascript
const queryClient = new QueryClient({
   defaultOptions: {
     queries: {
       refetchOnWindowFocus: false,
     },
   },
 })
```

- for local usage
```javascript
 useQuery('todos', fetchTodos, { refetchOnWindowFocus: false })
```

**Comment by TkDodo** [comment]
> I want to reload the component on window focus only when there is the change in data.

when refetching, the component transitions to `isFetching:true` and then back to false again. You can optimize re-renders with `notifyOnChangeProps`, but usually, it shouldn’t matter.
